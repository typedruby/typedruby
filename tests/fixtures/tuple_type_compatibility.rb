# @typedruby

def f([Integer, String] x) => nil; end

def test1 => nil
  f [1]
end

def test2 => nil
  f [1, "foo"]
end

def test3 => nil
  f [1, "foo", :bar]
end

def test4 => nil
  f ["foo", :bar, 1]
end
