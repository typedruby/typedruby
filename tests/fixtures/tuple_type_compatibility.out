info: Typechecking...
error: Could not match types:

        @ __ROOT__/tests/fixtures/tuple_type_compatibility.rb:3
      3 |  def f([Integer, String] x) => nil; end
                 ^^^^^^^^^^^^^^^^^ [Integer, String], with:
        @ __ROOT__/tests/fixtures/tuple_type_compatibility.rb:6
      6 |    f [1]
               ^^^ [Integer]
        @ __ROOT__/tests/fixtures/tuple_type_compatibility.rb:6
      6 |    f [1]
             ^^^^^ in this expression

error: Could not match types:

        @ __ROOT__/tests/fixtures/tuple_type_compatibility.rb:3
      3 |  def f([Integer, String] x) => nil; end
                 ^^^^^^^^^^^^^^^^^ [Integer, String], with:
        @ __ROOT__/tests/fixtures/tuple_type_compatibility.rb:14
     14 |    f [1, "foo", :bar]
               ^^^^^^^^^^^^^^^^ [Integer, String, Symbol]
        @ __ROOT__/tests/fixtures/tuple_type_compatibility.rb:14
     14 |    f [1, "foo", :bar]
             ^^^^^^^^^^^^^^^^^^ in this expression

error: Could not match types:

        @ __ROOT__/tests/fixtures/tuple_type_compatibility.rb:3
      3 |  def f([Integer, String] x) => nil; end
                  ^^^^^^^ Integer, with:
        @ __ROOT__/tests/fixtures/tuple_type_compatibility.rb:18
     18 |    f ["foo", :bar, 1]
                ^^^^^ String

        - arising from an attempt to match:

        @ __ROOT__/tests/fixtures/tuple_type_compatibility.rb:3
      3 |  def f([Integer, String] x) => nil; end
                 ^^^^^^^^^^^^^^^^^ [Integer, String], with:
        @ __ROOT__/tests/fixtures/tuple_type_compatibility.rb:18
     18 |    f ["foo", :bar, 1]
               ^^^^^^^^^^^^^^^^ [String, Symbol, Integer]
        @ __ROOT__/tests/fixtures/tuple_type_compatibility.rb:18
     18 |    f ["foo", :bar, 1]
             ^^^^^^^^^^^^^^^^^^ in this expression
